File: mobile/lib/features/home/domain/entities/exam_date.dart
import 'package:equatable/equatable.dart';

class ExamDate extends Equatable {
  final String id;
  final DateTime date;

  const ExamDate({
    required this.id,
    required this.date,
  });

  @override
  List<Object?> get props => [
        id,
        date,
      ];
}

File: new_backend/src/models/examDate.ts

import { Document, Schema, model } from "mongoose";

export interface IExamDate extends Document {
//   id: Schema.Types.ObjectId;
  date: String;
}

const ExamDateSchema: Schema<IExamDate> = new Schema({
  date:{
    type: String,
    required:[true,"Exam Date is required"]
  }
},
{
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt'
    }
  }
);

const ExamDate = model<IExamDate>("ExamDate", ExamDateSchema);

export default ExamDate;
